packages/notifications/src/error.ts

```diff
@@ -0,0 +1,87 @@
+import { AnyPromiseFn } from '@terra-ui-packages/core-utils';
+
+import { NotificationsContract, Notifier } from './useNotifications';
+
+export type ErrorCallback = (error: unknown) => void | Promise<void>;
+
+/**
+ * Should return true of a given error should be ignored
+ */
+export type IgnoreErrorDecider = (title: string, obj?: unknown) => boolean;
+
+/**
+ * Invoke the `callback` with any error thrown when evaluating the async `fn` with `...args`.
+ */
+export const withErrorHandling = (callback: ErrorCallback) => {
+  return <F extends AnyPromiseFn>(fn: F) => {
+    return (async (...args: Parameters<F>) => {
+      try {
+        return await fn(...args);
+      } catch (error) {
+        await callback(error);
+      }
+    }) as F;
+  };
+};
+
+/**
+ * Return a Promise to the result of evaluating the async `fn` with `...args` or undefined if
+ * evaluation fails.
+ */
+export const withErrorIgnoring = withErrorHandling(() => {});
+
+/**
+ * allows composition of NotificationsContract dependency when useful for code reuse
+ * @param notifier
+ */
+export const makeNotificationsContract = (
```

Comment:
Not keen on this name... we've been using "contract" for types, not concrete objects.